baremetalpxeserver/baremetalpxeserver/pxeserveragent.py

```
@@ -558,15 +587,25 @@ def _create_preconfiguration_file(self, cmd):
             rendered_content = self._render_autoyast_template(cmd, pxeserver_dhcp_nic_ip)
+        elif cmd.preconfigurationType == 'autoinstall':
+            rendered_content = self._render_autoinstall_template(cmd, pxeserver_dhcp_nic_ip)
         else:
             raise PxeServerError("unkown preconfiguration type %s" % cmd.preconfigurationType)
 
         ks_cfg_name = cmd.pxeNicMac
         ks_cfg_file = os.path.join(self.KS_CFG_PATH, ks_cfg_name)
-        with open(ks_cfg_file, 'w') as f:
-            f.write(rendered_content)
+        if cmd.preconfigurationType == 'autoinstall':
+            bash_r("mkdir -p %s" % ks_cfg_file)
```

Ruff (0.8.2)

`bash_r` may be undefined, or defined from star imports

(F405)